code/modules/antagonists/_common/antag_datum.dm

```diff
@@ -355,6 +355,11 @@ GLOBAL_LIST_EMPTY(antagonists)
 /datum/antagonist/proc/finalize_antag()
 	return
 
+/datum/antagonist/proc/admin_add(datum/mind/new_owner, mob/admin)
+	message_admins("[key_name_admin(admin)] made [key_name_admin(new_owner)] into [name].")
+	log_admin("[key_name(admin)] made [key_name(new_owner)] into [name].")
+	new_owner.add_antag_datum(src)
+
```

Comment:
Ify on adding antag datums inside of itself like this. Also you're going to need to ensure that if you do
```dm
var/antagonist/whatever/A = new
A.admin_add(src)
```
And `admin_add` returns without adding the datum, that you delete A, because it's already gone through it's `/New` at that point, and has been added to `GLOB.antagonists`. Thought this can probably be solved by putting the `GLOB.antagonists += src` into `on_gain()`. That probably makes more sense than being in New.

Reply:
No issue found in testing on the proc itself.

I've added a check for the admin_add call, the antag datum is qdel'd if admin_add does not return TRUE on succesful antag creation.